ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -780,6 +780,20 @@ module API {
       or
       pos.isBlock() and
       result = Label::blockParameter()
+      or
+      pos.isAny() and
+      (
+        result = Label::parameter(_)
+        or
+        result = Label::keywordParameter(_)
+        or
+        result = Label::blockParameter()
+        // NOTE: `self` should NOT be included, as described in the QLDoc for `isAny()`
+      )
+      // TODO: needs handling of `self` ArgumentPosition
```

Comment:
FWIW it was deliberate that I didn't include `self` here. I didn't see a use-case for it in Ruby API graphs, since in practice it means going back to where you came from. E.g. `base.getMethod("foo").getSelf()` would just be `base`. It's possible we'll need it for marking the post-update node of `self` as a taint source, I'm not sure.

(JS is different since any value can be passed as `this`, and some libraries take advantage of this)

Reply:
```suggestion
      // Note: it is deliberate that `self` ArgumentPosition is not included here.
      // I (@asgerf) didn't see a use-case for it in Ruby API graphs, since in practice
      // it means going back to where you came from. E.g. base.getMethod("foo").getSelf() 
      // would just be base. It's possible we'll need it for marking the post-update node 
      // of self as a taint source, I'm not sure.
```

Reply:
Sorry, there's no need to change this, I just wanted to explain why `self` isn't handled. The above isn't a style of comment I'd recommend inserting in the codebase.

Reply:
Why not? I think it's useful context for the next person looking at this code, and certainly more informative than the TODO I have left right now.

Reply:
OK then, I've tried to reformulate it to something I'd be willing to commit:
```suggestion
      // Note: there is currently no API graph label for `self`.
      // It was omitted since in practice it means going back to where you came from.
      // For example, `base.getMethod("foo").getSelf()` would just be `base`.
      // However, it's possible we'll need it later, for identifying `self` parameters or post-update nodes.
```